### PR TITLE
Issues/custom field fls

### PR DESCRIPTION
**Pull request by kpsf**
Added endpoint to update the System Admin's fls permissions. Input is an array of fields.

### CODE REVIEW

File: src/main/java/com/salesforce/scmt/service/SalesforceService.java [hunk 1]
```diff
@@ -808,6 +841,27 @@ public static String createRemoteSite(Request req, Response res) throws Exceptio
...
+        try {
+            List<String> pl = new Gson().fromJson(req.body(), ArrayList.class);
+            SalesforceService sf = new SalesforceService(salesforceUrl, salesforceSessionId);
+            sf.updateFieldPermissions(pl);
+        } catch(com.sforce.ws.SoapFaultException e) {
+            if (e.getMessage().contains("INVALID_SESSION_ID")) {
+                res.status(401);
+                return "Unauthorized";
+            }
+        } catch(Exception e) {
```
Comment (tstachl) [inline src/main/java/com/salesforce/scmt/service/SalesforceService.java]: We might want to remove this, to actually have errors logged.

File: src/main/java/com/salesforce/scmt/service/SalesforceService.java [hunk 2]
```diff
@@ -822,7 +876,7 @@ public static String createDataCategoryGroup(Request req, Response res) throws E
                 return "Unauthorized";
             }
         } catch(Exception e) {
-            res.status(200);
+            res.status(400);
```
Comment (tstachl) [inline src/main/java/com/salesforce/scmt/service/SalesforceService.java]: This doesn't fall into the scope of the issue, plus we should think of a way of throwing exceptions to have some error logging.

File: src/main/java/com/salesforce/scmt/utils/DeskJsonMapUtil.java [hunk 3]
```diff
@@ -171,7 +171,7 @@ else if (cf.getData().getType() == CustomFieldDataType.INTEGER && deskCFs.get(ke
         // Map status values from Desk.com to Salesforce
         mapObj.put(CaseFields.Status, (deskCase.getStatus() == CaseStatus.NEW ? CaseFields.StatusNew :
             (deskCase.getStatus() == CaseStatus.CLOSED ? CaseFields.StatusClosed :
-            (deskCase.getStatus() == CaseStatus.OPEN ? CaseFields.StatusWorking :
+            (deskCase.getStatus() == CaseStatus.OPEN ? CaseFields.StatusOpen :
```
Comment (tstachl) [inline src/main/java/com/salesforce/scmt/utils/DeskJsonMapUtil.java]: Do we always have a status open now in SFDC?

Comment (kpsf) [reply src/main/java/com/salesforce/scmt/utils/DeskJsonMapUtil.java]: These status changes are already merged into master so not sure why they are showing up here:

https://github.com/forcedotcom/scmt-server/pull/90/files

File: src/main/java/com/salesforce/scmt/service/SalesforceService.java (hunk 2 above)
Comment (kpsf) [reply src/main/java/com/salesforce/scmt/service/SalesforceService.java]: Tried to sneak this one in there...

**Comment by kpsf** [comment]
Do we need to fix those status changes or is it good? @tstachl

**Comment by tstachl** [comment]
@kpsf check this out, merging master get's rid of that change 🍔

**Pull request merged**